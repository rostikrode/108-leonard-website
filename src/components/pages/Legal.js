import React, {Component} from 'react';
import MultilineText from '../partials/MultilineText';
import ScrollArrow from '../partials/ScrollArrow';
import '../../styles/Legal.css';

export default class Legal extends Component {
  constructor(props) {
    super(props);
    
    this.onTabClick = this.onTabClick.bind(this);
  }
  componentDidMount() {
    /** meta data for page */
    document.title = this.props.metaTitle;
    if(document.getElementsByTagName('meta').description) {
      document.getElementsByTagName('meta').description.content = this.props.metaDescription
    }
    if (document.querySelector("link[rel='canonical']")) {
      document.querySelector("link[rel='canonical']").href = window.location.href
    }
    var viewport = document.querySelector("meta[name=viewport]");
    if(viewport) {
      viewport.setAttribute('content', 'width=device-width, initial-scale=1, user-scalable=1');
    }  
  }

  onTabClick(e) {
    console.log(e.currentTarget.dataset.id)

    var tabs = this.tabref.querySelectorAll('.tab');
    for (var t = 0; t < tabs.length; t++) {
      tabs[t].classList.remove('active');
    };
    var contents = this.contentref.querySelectorAll('.content');
    for (var c = 0; c < contents.length; c++) {
      contents[c].classList.remove('active');
    };

    e.currentTarget.classList.toggle('active');
    this.contentref.querySelector(`.content[data-id="${e.currentTarget.dataset.id}"]`).classList.toggle('active');

    this.scrollArrow.hideShowDownArrow();
  }

  render() {
    return (
      <div className="legal-wrapper">
        <div ref={e => {this.tabref = e}} className="tab-wrapper">
          {this.props.body.map((tab, key) => {
            return (
              <button key={key} onClick={this.onTabClick} className={key === 0 ? 'tab sans active' : 'tab sans'} data-id={`tab-${key}`}>{tab.tab}</button>     
            );
          })}
        </div>
        <div ref={e => {this.contentref = e}} className="content-wrapper">
          {this.props.body.map((content, key) => {
            return (
              <div key={key} className={key === 0 ? 'content sans active' : 'content sans'} data-id={`tab-${key}`}>
                <MultilineText text={content.content}/>
              </div>     
            );
          })}
        </div>
        <ScrollArrow ref={i => {this.scrollArrow = i;}} listElementRef={this.contentref} />
      </div>
    );
  }
}