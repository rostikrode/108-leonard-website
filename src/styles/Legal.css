.legal-wrapper {
  padding: 18px;
  height: calc(100vh - 36px - 90px);
}
@media (min-width: 1024px) {
  .legal-wrapper {
    padding: 55px;
    height: calc(100vh - 110px);
  } 
}

.tab-wrapper {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-around;
  align-items: center;
  padding: 18px 0;
  width: 100%;
  position: fixed;
  top: 80px;
  background-color: #fff;
}
@media (min-width: 1024px) {
  .tab-wrapper {
    position: relative;
    top: unset;
    padding: 30px 0;
    justify-content: flex-start;
  }
}

.tab-wrapper .tab {
  padding: 18px; padding-left: 0;
  border-radius: 0;
  border: 0;
  box-shadow: none;
  background-color: transparent;
  cursor: pointer;
  font-size: 13px;
}
.tab-wrapper .tab:hover {
  font-weight: 300;
}
.tab-wrapper .tab.active {
  font-weight: 600;
}
@media (min-width: 1024px) {
  .tab {
    font-size: 15px;
  }
}

.content-wrapper {  
  padding-top: 90px;
}
@media (min-width: 1024px) {
  .content-wrapper {
    padding-top: 0;
    overflow-y: auto;
    height: calc(100% - 110px);
  } 
}
.content-wrapper .content {
  display: none;
  text-align: left;
  font-size: 15px;
  line-height: 25px;
  letter-spacing: 1px;
  text-transform: none;
}
.content-wrapper .content.active {
  display: block;
}