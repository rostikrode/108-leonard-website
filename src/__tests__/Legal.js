import 'babel-polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Legal from '../components/pages/Legal';
import legalJSON from '../components/data/legal.json';
import { MemoryRouter } from 'react-router-dom';

window.gtag = window.gtag || function() {
  return true;
};

it('renders without crashing', () => {
  const tree = renderer.create(
    <MemoryRouter><Legal {...legalJSON}  /></MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});