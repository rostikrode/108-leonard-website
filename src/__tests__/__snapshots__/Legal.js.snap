// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="legal-wrapper"
>
  <div
    className="tab-wrapper"
  >
    <button
      className="tab sans active"
      data-id="tab-0"
      onClick={[Function]}
    >
      Privacy Policy
    </button>
    <button
      className="tab sans"
      data-id="tab-1"
      onClick={[Function]}
    >
      Legal Disclaimer
    </button>
    <button
      className="tab sans"
      data-id="tab-2"
      onClick={[Function]}
    >
      Image Credits
    </button>
  </div>
  <div
    className="content-wrapper"
  >
    <div
      className="content sans active"
      data-id="tab-0"
    >
      <div>
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "CIVIC CENTER COMMUNITY GROUP BROADWAY LLC PRIVACY POLICY",
            }
          }
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "\\"Civic Center Community Group Broadway LLC\\" (referred to herein as \\"108 Leonard\\", \\"we\\" or \\"us\\") is committed to protecting the privacy of the users of its website. This statement discloses our privacy practices. The purpose of this statement is to inform you:",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "What kinds of information we collect from users of \\"web address\\" and how the information is used by us",
            }
          }
          style={
            Object {
              "marginLeft": "30px",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Whether we disclose any user information to third parties;",
            }
          }
          style={
            Object {
              "marginLeft": "30px",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "How you can access, update or delete any information that we collect about you; and",
            }
          }
          style={
            Object {
              "marginLeft": "30px",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "The security procedures in which we use to protect your personal information",
            }
          }
          style={
            Object {
              "marginLeft": "30px",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "By using the Website, you’re agreeing to the 108 Leonard Privacy Policy. We reserve the right to change, modify, add, or remove portions of our Privacy Policy at any time. Please check this page periodically for changes.",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "If you have any questions about our Privacy Policy, please e-mail us at info@108leonard.com",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "How We Collect Information",
            }
          }
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "You can visit 108leonard.com without revealing any personal information. Our web servers collect the Internet Protocol addresses, but not the e-mail addresses, of visitors. This information can measure the number of visits, average time spent on the site, pages viewed, and other such statistics.",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "In addition, You may be asked for information that identifies you, including your name, e-mail address, mailing address, zip code, telephone number, fax (collectively, \\"Personal Information\\"). You may also elect to provide such information to us by sending or responding to e-mails regarding certain listings or services offered on our Website.",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "How We Use and Disclose Your Information",
            }
          }
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "We collect, generate, retain and use your Personal Information for our own internal purposes in connection with the facilitation, recording and processing of any requests, communications, or interactions you may have with our Website. We also automatically collect and store statistics and other information about you and your online activities on an aggregated, non-personally identifiable basis and in a manner that may allow us or our affiliated or related entities to improve our services to you.",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "In addition, your Personal Information may be also be used by us to provide you with information regarding our products and services. For example, we may use your e-mail address to send you special announcements and notifications of new real estate listings, services or promotions that may be of interest to you.",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Cookies",
            }
          }
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Cookies allow websites to recognize you when you return and can keep track of information specific to you. Consistent with standard practices in the Internet industry, you may occasionally receive cookies from third parties to which you link from our Website. If you prefer, you may set your browser to ask for your permission before your receive a cookie. We do not control these third party cookies, and they are not subject to this Privacy Policy.",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Information Sharing and Disclosure",
            }
          }
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "When you request information, we need to know your name, e-mail address, and mailing address. This allows us to process and fulfill your request. 108leonard.com does not sell, trade, or rent your personal information to others 108leonard.com may provide aggregate statistics about our customers, sales, traffic patterns, and related site information to reputable third-party vendors, but these statistics will include no personally identifying information.",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Data Security",
            }
          }
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Data security is critical to us and thus all Personal Information is held in a secured database. While it is impossible to guarantee the complete security of any computer system and the data contained therein, our vendors are required to maintain and implement robust security policies and procedures that combine with available technologies in accordance with prevailing industry standards, all of which are designed to protect the confidentiality, integrity, and availability of your Personal Information. To the extent we are provided with social security numbers or personal financial information, we comply with all applicable regulations regarding the confidentiality and safe disposal of such information.",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Accessing, Updating or Deleting Information Collected About You by 108 Leonard",
            }
          }
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "If you wish to access, update, or delete contact information or preferences, please send an e-mail to info@108leonard.com. Please be aware, however, that we cannot always ensure that such corrections or deletions will immediately be made in our database.",
            }
          }
        />
      </div>
    </div>
    <div
      className="content sans"
      data-id="tab-1"
    >
      <div>
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "All square footages are approximate. Development managed by Elad Group. Sponsor: Civic Center Community Group Broadway LLC (C3GB). The complete offering terms are in an offering plan available from Sponsor File # CD16-0364. Sponsor: Civic Center Community Group Broadway LLC, having an address c/o El Ad US Holding, Inc., 575 Madison Avenue, 22nd Floor, New York, New York 10022. Units will not be offered furnished or with landscaping. Furniture and landscaping layouts shown are for concept only and are not coordinated with building systems. No representation or warranty is made that a unit owner will be able to implement the furniture or landscaping layout shown on this floor plan. Due to the unique fa&ccedil;ade of this building, unit area per floor as well as windows in this layout may vary from floor to floor. Refer to the offering plan for dimensions and conditions. Equal Housing Opportunity. DBOX",
            }
          }
        />
      </div>
    </div>
    <div
      className="content sans"
      data-id="tab-2"
    >
      <div>
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "All images, design and text by DBOX, unless noted.",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Aire Baths, courtesy of Aire Ancient Baths",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "American Cut, courtesy of American Cut",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Le Coucou, Ditte Isager, courtesy of Le Coucou",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Todd Merrill, courtesy of Todd Merrill",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "De Vera, courtesy of De Vera",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Taryn Toomey, Jaimie Baird, courtesy of Taryn Toomey",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Ascent Lounge, Steven Menendez, courtesy of Jeffrey Beers International",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Seared Restaurant, courtesy of Jeffrey Beers International",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "The Cove Atlantis, courtesy of Jeffrey Beers International",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "One West End, Matt Simpkins Photography, courtesy of Elad",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "22 Central Park South, Evan Joseph, courtesy of Elad",
            }
          }
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "250 West Street, Evan Joseph, courtesy of Elad",
            }
          }
        />
      </div>
    </div>
  </div>
  <img
    alt="downward arrow icon"
    className="arrow-down-scroll"
    src="down_arrow_large.svg"
  />
</div>
`;
